stripe/resource_stripe_price.go

```diff
@@ -556,14 +562,16 @@ func resourceStripePriceCreate(ctx context.Context, d *schema.ResourceData, m in
 		recurringMap := ToMap(recurring)
 		for k, v := range recurringMap {
 			switch {
-			case k == "interval" && ToString(v) != "":
+			case k == "interval":
 				params.Recurring.Interval = stripe.String(ToString(v))
 			case k == "interval_count" && ToString(recurringMap["interval"]) != "":
 				params.Recurring.IntervalCount = stripe.Int64(ToInt64(v))
-			case k == "aggregate_usage" && ToString(v) != "":
```

Comment:
Please revert this one - When the item is not required we don't want to call the Stripe API with the Go zero value (in this case empty string). The Terraform populates the map keys even there are no values which is causing this behavior.

The rule of thumb here is:

1. `case k == "interval"`-> the interval is required field therefore no need to check for empty value (when value is empty the API responses with error)
2. `case k == "aggregate_usage" && ToString(v) != ""` -> the aggregate_usage is not required therefore check is needed. 

Does it make more sense:?

Reply:
yes, thanks!